libcxx/test/std/algorithms/alg.nonmodifying/alg.find/find.pass.cpp

```diff
@@ -113,6 +113,272 @@ struct TestTypes {
   }
 };
 
+enum class match {
+  none,
+  first,
+};
+
+template <class T, class U>
+TEST_CONSTEXPR_CXX20 void test_integer_promotion(match min_expected, match max_expected) {
+  { // check min value
```

Comment:
Instead you could say:

```
bool should_match = T(value) == U(value);
```

and then it doesn't really matter what size different types are, you'd be checking that we have the same behavior that a "naive" implementation of `std::find` (using `==`) would have, which is really what we care about here.

Reply:
And then you can throw a bunch of different values at the test much more easily, I think.